primitives/core/Cargo.toml

```diff
@@ -51,6 +51,9 @@ schnorrkel = { version = "0.9.1", features = [
 	"preaudit_deprecated",
 	"u64_backend",
 ], default-features = false, optional = true }
+bls-like = {git = "https://github.com/w3f/bls", default-features = false}
```

Comment:
Imho, we should not depend upon the `EngineBLS` trait here.  It's completely useless.  It'll also cause issues for portability and maintenance.

Reply:
bls crypto type within substrate has been defined to be generic. 
This allowed to easily have both bls12-381 and bls12-377 with the minimum effort and code duplication.
To have such kind of generic we bounded the generic type to be an `EngineBLS` trait.
AFAIK is the only bound that allow us to do provide a generic implementation.
Is there some other trait in `bls-like` crate allowing this?

Reply:
We should not imho use the bls-like crate in production.  I started it as an experiment, and much has been added, but it was never cleaned up, so much of what is there no longer serves any purpose, and is generally a confusing poorly structured mess.   https://github.com/w3f/ring-vrf/tree/master/nugget_bls was written to do exactly what beefy requires.

Reply:
I made this repo https://github.com/w3f/BCLS-BLS I'll start isolating the bls logic from nugget and bls-like to here and eventually relpace the backend. 

Nonetheless I don't want this to become a blocker on this pull requset as, given that the output of the BLS-like matches the reference in the paper, BEEFY's integration is blocked on this, production ready BLS BEEFY is also blocked on apk proof adoption of BW6 on BLS12-381 and we are exposing no host functio. 

Therefore, I don't think blocking the pull request due to some opinion about an abstraction in the backend is justified.

Reply:
Or just use nugget_bls after changing whatever needs changing?  At minimum it helps consolidate our code which should be hardened for non-constant-time elliptic curves. 

I've found an unrelated annoyance with merlin, so I'll do some ark_transcript crate based on shake128 btw.